dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaAPIHostsTask.java

```diff
@@ -63,14 +64,20 @@ protected void doRun(
         JsonNode jsonHosts;
         try (CloseableHttpResponse hostsResponse =
             httpClient.execute(new HttpGet(hostPerClusterUrl))) {
-          jsonHosts = getObjectMapper().readTree(hostsResponse.getEntity().getContent());
+          final int statusCode = hostsResponse.getStatusLine().getStatusCode();
+          if (!isStatusCodeOK(statusCode)) {
+            throw new MetadataDumperUsageException(
+                "Cloudera Error: Response status code is not 2xx.");
```

Comment:
You lose information here. You just throw and exception but not set or log the incorrect data, in this case it is `statusCode` or full `hostsResponse`.
Error message can be something like: 'Cloudera API response is [actual] but expected 2xx`.

Reply:
Updated the error message.